sound/soc/sof/pcm.c

```diff
@@ -470,6 +470,10 @@ static int sof_pcm_trigger(struct snd_soc_component *component,
 		if (ret < 0)
 			return ret;
 
+		ret = snd_sof_pcm_platform_hw_free(sdev, substream);
+		if (ret < 0)
+			return ret;
```

Comment:
I am not able to review this without having to look at code, this is really painful with function names that look nearly identical, can we please add comments to highlight the differences between sof_pcm_dsp_pcm_free() and snd_sof_pcm_platform_hw_free(), and why the order matters.

Reply:
The commit message is also convoluted

"To enable restarting the stream without a hw_free"
Which hw_free are we talking about? In the existing sequence, on TRIGGER_STOP, we also send an IPC_HW_FREE. But in the 'regular' ALSA sequences, hw_free is never invoked with a stop.